handlers/handlers.go

```diff
@@ -0,0 +1,104 @@
+package handlers
+
+import (
+	"blog/helper"
+	"blog/models"
+	"strconv"
+	"time"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/bcrypt"
+)
+
+// This is for Signup
+func Signup(c *fiber.Ctx) error {
+	var data models.User
+	check := false
+
+	//Get user details from request body
+	if err := c.BodyParser(&data); err != nil {
+		return err
+	}
+	if data.Email == "" || data.Id <= 0 || data.Password == "" || data.Username == "" || data.Role == "" {
+		return c.JSON(fiber.Map{
+			"message": "missing credentials",
+		})
+	}
+
+	for _, val := range models.Database {
+		if val.Id == data.Id || val.Email == data.Email {
```

Comment:
Why if happen when the Id is correct and Email is wrong

Reply:
Here, we have to check only the Email is exist or not. Id validation is not necessary. So I will ignore that id validation.

Reply:
Then remove it